src/features/bookOffer/components/BookingDetails.native.test.tsx

```diff
@@ -78,161 +83,129 @@ jest
+    })
 
-  it('should dismiss modal on successfully booking an offer', async () => {
-    server.use(
-      rest.post(`${env.API_BASE_URL}/native/v1/bookings`, (req, res, ctx) => res(ctx.status(204)))
-    )
+    it('should initialize the state when offer isDigital only with first bookable stocks', async () => {
+      mockBookingStock = undefined
 
-    const { getByText } = renderBookingDetails(mockStocks)
+      await renderBookingDetails([{ ...offerStockResponseSnap, isBookable: false, id: 123456 }])
+      expect(mockDispatch).not.toHaveBeenCalled()
 
-    const ConfirmButton = getByText('Confirmer la réservation')
-    fireEvent.press(ConfirmButton)
+      await renderBookingDetails([
+        { ...offerStockResponseSnap, isBookable: false, id: 123456 },
+        { ...offerStockResponseSnap, isBookable: true, id: 1234567 },
+        { ...offerStockResponseSnap, isBookable: true, id: 12345678 },
+      ])
+      expect(mockDispatch).toHaveBeenCalledWith({ type: 'SELECT_STOCK', payload: 1234567 })
+      expect(mockDispatch).toHaveBeenCalledWith({ type: 'SELECT_QUANTITY', payload: 1 })
+      expect(mockDispatch).not.toHaveBeenCalledWith({ type: 'SELECT_STOCK', payload: 12345678 })
+    })
 
-    await waitFor(() => {
-      expect(mockDismissModal).toHaveBeenCalledTimes(1)
-      expect(analytics.logBookingConfirmation).toHaveBeenCalledWith(
-        mockOfferId,
-        undefined,
-        undefined
-      )
-      expect(campaignTracker.logEvent).toHaveBeenCalledWith(CampaignEvents.COMPLETE_BOOK_OFFER, {
-        af_offer_id: mockOffer.id,
-        af_booking_id: mockBookingStock?.id,
-        af_price: mockBookingStock?.price,
-        af_category: mockOffer.subcategoryId,
-      })
-      expect(navigate).toHaveBeenCalledWith('BookingConfirmation', { offerId: mockOfferId })
+    it('should not display the Duo selector when the offer is not duo', () => {
+      const { queryByTestId } = renderBookingDetails(mockDigitalStocks)
+
+      expect(queryByTestId('DuoChoiceSelector')).toBeNull()
     })
   })
 
-  it('should log the origin offer when booked an offer from similar offers', async () => {
-    useRoute.mockReturnValueOnce({
-      params: { fromOfferId: 1 },
+  describe('when user has selected options', () => {
+    beforeEach(() => {
+      mockUseBookingContext.mockReturnValueOnce({
+        bookingState: { quantity: 1, offerId: mockOfferId } as BookingState,
+        dismissModal: mockDismissModal,
+        dispatch: mockDispatch,
+      })
     })
-    server.use(
-      rest.post(`${env.API_BASE_URL}/native/v1/bookings`, (req, res, ctx) => res(ctx.status(204)))
-    )
 
-    const page = render(
-      // eslint-disable-next-line local-rules/no-react-query-provider-hoc
-      reactQueryProviderHOC(<BookingDetails stocks={mockStocks} />)
-    )
-
-    fireEvent.press(page.getByText('Confirmer la réservation'))
+    it('should render correctly when user has selected options and offer is an event', async () => {
```

Comment:
`should render correctly when offer is an event`

tu utilises `when user has selected options` dans ton `describe`